### PR TITLE
Update http-parser to 2.6.2

### PR DESCRIPTION
**Pull request by wildart**
This is update of `http-parser` version to 2.5.0 for Julia 0.4 only.

### CODE REVIEW

**Comment by IainNZ** [comment]
I don't know if we can really go 0.4 only yet

**Comment by wildart** [comment]
Current version works fine with 0.3, so we go with this PR when 0.4 is released.

**Comment by IainNZ** [comment]
OK

**Comment by yuyichao** [comment]
Ping. 0.4 is release now.

**Comment by wildart** [comment]
Ready.

File: src/HttpParser.jl [hunk 1]
```diff
@@ -66,6 +67,10 @@ Parser() = Parser(
     (global id_pool += 1)
 )
 
+# Datatype Tuples for the different `cfunction` signatures for callback functions
+HTTP_CB      = (Int, (Ptr{Parser},))
+HTTP_DATA_CB = (Int, (Ptr{Parser}, Ptr{Cchar}, Csize_t,))
+
```
Comment (yuyichao) [inline src/HttpParser.jl]: Is there a reason these are not `const`s?

Comment (wildart) [inline src/HttpParser.jl]: nope

File: deps/build.jl [hunk 2]
```diff
@@ -14,20 +14,19 @@ end
...
-    target = joinpath(prefix,"lib/libhttp_parser.$(BinDeps.shlib_ext)")
+    prefix = BinDeps.usrdir(libhttp_parser)
+    target = "libhttp_parser.$(BinDeps.shlib_ext)"
+    targetpath = joinpath(BinDeps.libdir(libhttp_parser),target)
 
     provides(SimpleBuild,
         (@build_steps begin
-            ChangeDirectory(Pkg.Dir.path("HttpParser"))
+            ChangeDirectory(BinDeps.pkgdir(libhttp_parser))
             FileRule("deps/src/http-parser/Makefile",`git submodule update --init`)
```
Comment (tkelman) [inline deps/build.jl]: better to just download a normal tarball here

**Comment by yuyichao** [comment]
Anything still blocking this?

**Comment by yuyichao** [comment]
Ping again?

**Comment by IainNZ** [comment]
@yuyichao, want commit access to JuliaWeb?

**Comment by yuyichao** [comment]
Sure ... =)

**Comment by yuyichao** [comment]
I assume it is ok to merge this and will do in a few days. The tarball vs git issue is independent of this PR and shouldn't hold this back.

**Comment by tkelman** [comment]
No I disagree, git submodules are unacceptable going forward. Git is no longer a runtime dependency of Julia.

**Comment by wildart** [comment]
I removed submodule, so C library is built from source archive. Ready to merge.

**Comment by tkelman** [comment]
Did the API change enough that the windows binaries need updating for this to still work?

**Comment by wildart** [comment]
Yes, there were some changes. We better build new win binaries.

**Comment by malmaud** [comment]
Thanks!

**Pull request merged**